### ISSUE TITLE
[feature] add `mle analyze` to summarize training logs and give optimal config

### ISSUE THREAD

**Issue by leeeizhang** [MLSysOps/MLE-agent#222]
RT

**Comment by huangyz0918** [comment]
Any updates on the issue?

**Comment by leeeizhang** [comment]
I think `mle analyze` is akin to an ML experiment summarizer and provides more tuning suggestions, such as hyper-parameters and NAS, based on the user's existing runs. To support this feature, we may need to integrate some ML tracking tools (e.g., wandb, mlflow, etc.) initially, and then build and test our ML tuner agents.

Additionally, the experimental summary could also be included in the report, as ML scientists may take a great interest in the weekly experimental progress. This task could be put on hold if any prior tasks take precedence.

- [ ] ML tracking integration (wandb)
- [ ] experiement summarizer
- [ ] weekly experiment reporter
- [ ] HPO and NAS suggestor

Refer to:
- https://arxiv.org/pdf/2402.01881
- https://arxiv.org/pdf/2309.01125
- https://arxiv.org/pdf/2302.14838
- https://github.com/automl/CAAFE

**Comment by huangyz0918** [comment]
What about integrating W&B first? By fetching the experiments, we can know the metrics, the code and the hyper parameters, etc. Then we use an agent to give suggestions, also calling some web search functions (e.g., paper with code benchmark) to give summary and suggestion (e.g., how to tune the parameters to improve certain metrics)

**Comment by huangyz0918** [comment]
The train/validation accuracy, the loss items are time-series data, how to analyze such data using LLM is also an interesting problem. Or we can fetch the visualization from W&B and try to analyze the images directly using the muti-modal ability, you can have a try and then we can discuss.

**Comment by leeeizhang** [comment]
> What about integrating W&B first? By fetching the experiments, we can know the metrics, the code and the hyper parameters, etc. Then we use an agent to give suggestions, also calling some web search functions (e.g., paper with code benchmark) to give summary and suggestion (e.g., how to tune the parameters to improve certain metrics)

That's right, I will integrate W&B first. For the analysis parts, we can incorporate the existing AdviseAgent to summarize and suggest.

> The train/validation accuracy, the loss items are time-series data, how to analyze such data using LLM is also an interesting problem. Or we can fetch the visualization from W&B and try to analyze the images directly using the muti-modal ability, you can have a try and then we can discuss.

agree with you! How to analyze time-series data is still an open question for exploration, and using multi-modal ability to directly analyze experimental plots or charts is very valuable to try. Nevertheless, since the most common NAS and NPO algorithms still use the final & best accuracy/loss for analyzing and tuning, we may also keep the possibility that directly using each run's best/final metrics as prompts to build the agent in our very beginning PoC.
